cranelift/codegen/src/isa/aarch64/lower.isle

```diff
@@ -1955,6 +1955,16 @@
 (rule (lower (trap trap_code))
       (side_effect (udf trap_code)))
 
+;;;;;  Rules for `trapz`;;;;;;;;;
+
+(rule (lower (trapz val trap_code))
+  (trap_if_zero $true val trap_code))
+
+;;;;;  Rules for `trapnz`;;;;;;;;;
+
+(rule (lower (trapnz val trap_code))
+  (trap_if_zero $false val trap_code))
```

Comment:
Can we rename `trap_if_zero` to something like `trap_if_reg`? Having `(trap_if_zero $false ...)` mean that we are trapping if something is *not* zero is a bit too easy to mix up. Even better would be to define a custom enum for zero and not zero because boolean arguments to functions almost always make things harder to read than otherwise.

Reply:
Renamed and used an enum